src/algebra/algebra/basic.lean

```diff
@@ -348,6 +348,15 @@ instance of_subsemiring (S : subsemiring R) : algebra S A :=
   smul_def' := λ r x, algebra.smul_def r x,
   .. (algebra_map R A).comp S.subtype }
 
+lemma algebra_map_of_subsemiring {R : Type*} [comm_semiring R] (S : subsemiring R) :
+  (algebra_map S R : S →+* R) = subsemiring.subtype S := rfl
+
+lemma coe_algebra_map_of_subsemiring {R : Type*} [comm_semiring R] (S : subsemiring R) :
+  (algebra_map S R : S → R) = subtype.val := rfl
+
+lemma algebra_map_of_subsemiring_apply {R : Type*} [comm_semiring R] (S : subsemiring R) (x : S) :
+  algebra_map S R x = x := rfl
```

Comment:
I don't think you need the `{R : Type*} [comm_semiring R]` here, that's already present in this section (the `subring` case below is different)